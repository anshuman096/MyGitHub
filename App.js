import React, { Component } from 'react';
import { ScrollView, 
	Text, 
	TextInput, 
	View, 
	TouchableOpacity, 
	Image, 
	ImageBackground, 
	ActivityIndicator, 
	AsyncStorage, 
	StyleSheet, 
	Alert} from 'react-native';
const base64 = require('base-64'); 
import MyGitHubNavigator from './routes/MyRoutes';

export default class App extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            isLoggedIn: false,
            password: '',
            isLoading: false,
            authToken: ''
        }
    }

    async saveData(key, value) {
    	await AsyncStorage.setItem(key, value);
    }
    
    _userLogin = () => {
        this.setState({isLoading: true});
        this.state.authToken = 'Basic ' +  base64.encode(this.state.username + ':' + this.state.password);
        this.saveData('authToken', this.state.authToken);
        this.saveData('currentUser', this.state.username);
        this.saveData('handle', this.state.username); //for first login
        fetch('https://api.github.com/user', {
			method: 'GET',
            headers: {
                'Content-Type':'application/json',
                'Authorization':this.state.authToken
            }
		}).then((response) => {
                this.setState({isLoading: false});
                if(response.status == 200)
                    this.setState({isLoggedIn: true});    
                else 
                    Alert.alert('Invalid login');
			}).catch((error) => {
			     console.error(error);
		});
    }
    
    
    render() {
        //isLoggedIn is true
        if(this.state.isLoggedIn) {
            return (<MyGitHubNavigator
                //onLogoutPress={() => this.setState({isLoggedIn: false})}
            />);
            //
        } else {
            return (
            <ImageBackground source={require('./images/background.jpg')} style = {styles.backgroundImage} resizeMode={Image.resizeMode.stretch}>
                <ScrollView style={styles.mainContainer}>
                    <View style = {styles.titleContainer}>
                        <Text style={styles.titleLabel}> Git My Hub </Text>
                        <Image source={require('./images/github-512.png')} style = {styles.avatar}/>
                    </View>
                    <View style = {styles.loginContainer}>
                        <TextInput style = {styles.loginLabel} 
                            selectTextOnFocus
                            placeholder='Username' 
                            autoCapitalize = 'none'
                            onChangeText = {(text) => this.setState({username: text})}
                            blurOnSubmit = {true}
                        />
                        <TextInput style = {styles.loginLabel} 
                            placeholder='Password' 
                            selectTextOnFocus 
                            autoCapitalize = 'none'
                            onChangeText = {(text) => this.setState({password: text})}
                            secureTextEntry = {true}
                            onEnter = {this._userLogin}
                        />
                        <View style={{margin:7}} />
                        <TouchableOpacity 
                            style = {styles.viewButton} 
                            disabled = {this.state.isLoggedIn||!this.state.username||!this.state.password} 
                            onPress = {this._userLogin}>
                                <Text style = {styles.buttonText}>Login</Text>
                        </TouchableOpacity>
                        <View style={{margin:7}} />
                        <ActivityIndicator size = 'large' color = '#0000ff' animating = {this.state.isLoading} hidesWhenStopped = {true}/>
                    </View>
                </ScrollView>
            </ImageBackground>
        );
        }
    }
}

const styles = StyleSheet.create({
    mainContainer: {
        flex: 1,
        padding: 20
    },
    titleContainer: {
        flex: 20,
        alignItems: 'center'
    },
    loginContainer: {
        flex: 2
    },
    backgroundImage: {
        flex: 1
    },
    avatar: {
        height: 200,
        width: 200,
        paddingBottom: 50
    },
    titleLabel: {
        fontSize: 27,
        fontFamily: 'Avenir',
        fontWeight: 'bold',
        paddingBottom: 25
    },
    loginLabel: {
        alignItems: 'center',
        borderBottomWidth: 1,
        borderBottomColor: 'black',
        fontSize: 18,
        paddingTop: 20,
        borderRadius: 8
    },
    viewButton: {
        backgroundColor: '#f5f5f5',
        padding:10,
        borderRadius: 4
    },
    buttonText: {
        color: 'black',
        textAlign: 'center',
        fontSize: 20
    }
});

