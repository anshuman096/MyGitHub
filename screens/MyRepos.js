import React, { Component } from 'react';
import {View, FlatList, Text, StyleSheet, Button, Alert, Linking, AsyncStorage} from 'react-native';
import RepoList from './RepoList';


/**
 * The Repos Class for MyGitHub app. Lists all the public
 * repositories for the user.
 *
 * @author: Anshuman Dikhit
 */
export default class MyRepos extends Component {

	constructor(props) {
		super(props);
		this.state = {
			isLoading : true,
			data : '',
            username: '',
            authToken: ''
		}
	}
    

    async componentWillMount() {
		console.log("MyRepos....");
        var user = await AsyncStorage.getItem('handle');
        var authToken = await AsyncStorage.getItem('authToken');
        await this.setState({
            isLoading: false,
            username: user,
            authToken: authToken
        });
    }
    
    /**
     * render invokes a component called RepoList, implemented in a 
     * separate file.
     */
    render() {
		console.log("MyRepos....    2");
        if(this.state.isLoading == true)
            return null;
        
		console.log("MyRepos....    3");
        return(
            <View style = {styles.mainbackground}>
                <Text style = {styles.title}>Public Repositories</Text>
                <RepoList style = {styles.background} 
                    username = {this.state.username} 
                    authToken = {this.state.authToken}
                    navigation = {this.props.navigation}
				/>
            </View>
        );
    }  
}

const styles = StyleSheet.create({
    mainbackground: {
        backgroundColor: 'steelblue'
    },
    background: {
        backgroundColor: 'powderblue'
    },
    title: {
        fontSize: 25,
        paddingTop: 20,
        fontWeight: 'bold',
        fontFamily: 'Avenir',
        alignSelf: 'center',
    }
});
