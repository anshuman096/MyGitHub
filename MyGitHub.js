import React, {Component} from 'react';
import {View, FlatList, Image, Text, StyleSheet, Button, Alert, ScrollView, TouchableOpacity, Linking, AsyncStorage, ActivityIndicator} from 'react-native';
import {TabNavigator} from 'react-navigation';
import MyRepos from './Repos';
import MyFollowers from './Followers';
import MyFollowing from './Following';
const base64 = require('base-64'); //useful in creating a login page

/**
 * The Main Class for MyGitHub app. Provides basic data
 * on GitHub user, and has navigation to repository lists,
 * follower lists, and following lists.
 *
 * @author: Anshuman Dikhit
 */
class MyGitHub extends Component {

	constructor(props) {
		super(props);
		this.state = {
            mainUser: '',
            username: '',
            authToken: '',
			isLoading : true,
			data : ''
		}
	}
    
    
   /**
    * A method to be run before the render() function can be 
    * called. This method makes a GET ReST API call to GitHub with
    * the users handle and password (password is not secure atm), and returns
    * a JSON object with all the information on the user.
    *
    */
	async componentWillMount() {
        let currUser = await AsyncStorage.getItem('currentUser');
        let user = await AsyncStorage.getItem('handle');
        let authToken = await AsyncStorage.getItem('authToken');
        await this.setState({
            mainUser:currUser,
            username:user,
            authToken:authToken
        });

        var url = 'https://api.github.com/users/' + user;
        let response = await fetch(url, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': authToken
            }
        });

        let responseJson = await response.json();

        await this.setState({
            isLoading: false,
            data: responseJson
        });
        // store data in asynchronous storage
        await AsyncStorage.setItem('name', this.state.data.name);
        await AsyncStorage.setItem('handle', this.state.data.login);
        if(this.state.data.blog == '')
            await AsyncStorage.setItem('website', this.state.data.html_url);
        else
            await AsyncStorage.setItem('website', this.state.data.blog);
        await AsyncStorage.setItem('Date', this.state.data.created_at);
        await AsyncStorage.setItem('Bio', this.state.data.bio);
	}



    async shouldComponentUpdate(nextProps) {
        if(nextProps.navigation.state.params && nextProps.navigation.state.params.username) {
            // console.log('MyGitHub -> shouldComponentUpdate nextUser: ' + nextProps.navigation.state.params.username);
            // console.log('MyGitHub -> this.state.username: ' + this.state.username);
            if(this.state.username != nextProps.navigation.state.params.username) {
                //console.log('MyGitHub -> shouldComponentUpdate must re-render');
                await AsyncStorage.setItem('handle', nextProps.navigation.state.params.username);
                var url = 'https://api.github.com/users/' + this.state.username;

                let response = await fetch(url, {
                    method: 'GET',
                    headers: {
                        'Content-Type': 'application/json',
                        'Authorization': this.state.authToken
                    }
                });

                let responseJson = await response.json();

                await this.setState({
                    username: nextProps.navigation.state.params.username,
                    data: responseJson
                });
                return true;
            } else 
                return false;
        } else 
            return false;
        
    }
    
    /**
     * The render function for the main component. Displays title,
     * avatar, name, handle, email, bio, navigation to public repositories,
     * followers, and following
     * 
     */
	render() {
        let pic = {
            uri:this.state.data.avatar_url
        };
        
        // sets website to blog, if not blog then html_url
        var website = this.state.data.blog;
        if(this.state.data.blog == '')
            website = this.state.data.html_url;
            
        
        if(this.state.isLoading == true) {
            return(<ActivityIndicator size = 'large' color = '#0000ff' animating = {this.state.isLoading} />);
        }
        // placement of components for UI of main page
		return(
            <ScrollView style = {styles.mainContainer}>
                <View style = {{alignItems: 'flex-start', flexDirection: 'row'}}>
                    <Button onPress={() => {
                        console.log('MyGitHub render mainUser: ' + this.state.mainUser);
                        this.props.navigation.navigate('Home', {username: this.state.mainUser});}} title = 'Home'/>
                </View>
                <View style = {styles.titleContainer}>
                    <Image source = {pic} style = {styles.avatar}/>
                    <Text style = {styles.title}>{this.state.data.name}</Text>
                </View>
            
                <View style = {styles.userInfoContainer}>
            
                    <View style = {{flexDirection: 'row'}}>
                        <Text style = {styles.infoLabel} >Handle: </Text>
                        <Text style = {styles.infoLabel} >{this.state.data.login}</Text>
                    </View>
            
                    <View style = {{flexDirection: 'row'}}>
                        <Text style = {styles.infoLabel} >Website: </Text>
                        <Text style = {styles.infoLabel} >{website}</Text>
                    </View>

                    <View style = {{flexDirection: 'row'}}>
                        <Text style = {styles.infoLabel} >Date: </Text>
                        <Text style = {styles.infoLabel} >{this.state.data.created_at}</Text>
                    </View>

                    <View style = {{flexDirection: 'row'}}>
                        <Text style = {styles.infoLabel} >Email: </Text>
                        <Text style = {styles.infoLabel} >{this.state.data.email}</Text>
                    </View>

                    <View style = {{flexDirection: 'row'}}>
                        <Text style = {styles.infoLabel} >Bio: </Text>
                        <Text style = {styles.infoLabel} >{this.state.data.bio}</Text>
                    </View>

                </View>
                <View style = {{margin: 50}}/>
            </ScrollView>
        );
    }
}


// stylesheet for all placement and visual aspects of UI
const styles = StyleSheet.create({
    mainContainer: {
        flex: 1,
        paddingTop: 50
    }, 
    titleContainer: {
        flex: 3,
        alignItems: 'center',
        borderBottomWidth: 1,
        borderBottomColor: 'lightgrey'
    },
    editContainer: {
        alignItems: 'flex-end'  
    },
    userInfoContainer: {
        flex: 2,
        alignItems: 'flex-start'
    },
    avatar: {
        width: 250,
        height: 250,
        borderRadius: 125,
        paddingTop: 100
    },
    title: {
        fontSize: 30,
        fontWeight: 'bold',
        fontFamily: 'Avenir',
        paddingTop: 20
    },
    infoLabel: {
        fontSize: 18,
        paddingTop: 10,
        paddingBottom: 10,
        fontFamily: 'Avenir',
        color: '#a9a9a9'
    },
    modifyValue: {
        fontSize: 18,
        paddingTop: 10,
        paddingBottom: 10,
        fontFamily: 'Avenir',
        color: 'steelblue'
    }
});



//StackNavigator that determines navigation of all
//page within the app.
let routeConfigs = {
    Home: {
        screen: MyGitHub,
        navigationOptions: ({navigation}) => ({
            tabBarLabel: ({tintColor}) => <TouchableOpacity
                            onPress = {() => {navigation.navigate('Home', {date: new Date()});}}
                            style = {{alignItems: 'center', justifyContent: 'center'}}>
                            <Text style = {styles.infoLabel}>Profile</Text>
                        </TouchableOpacity>
        })
    },
    Repos: {
        screen: MyRepos,
        navigationOptions: ({navigation}) => ({
            tabBarLabel: ({tintColor}) => <TouchableOpacity
                            onPress = {() => {navigation.navigate('Repos', {date: new Date()});}}
                            style = {{alignItems: 'center', justifyContent: 'center'}}>
                            <Text style = {styles.infoLabel}>Repos</Text>
                        </TouchableOpacity>
        })
    },
    Followers: {
        screen: MyFollowers,
        navigationOptions: ({navigation}) => ({
            tabBarLabel: ({tintColor}) => <TouchableOpacity
                            onPress = {() => {
                                //console.log('MyGitHub navigation followers username: ' + this.state.data);
                                navigation.navigate('Followers', {date: new Date()});
                            }}
                            style = {{alignItems: 'center', justifyContent: 'center'}}>
                            <Text style = {styles.infoLabel}>Followers</Text>
                        </TouchableOpacity>
        })
    },
    Following: {
        screen: MyFollowing,
        navigationOptions: ({navigation}) => ({
            tabBarLabel: ({tintColor}) => <TouchableOpacity
                            onPress = {() => {navigation.navigate('Following', {date: new Date()});}}
                            style = {{alignItems: 'center', justifyContent: 'center'}}>
                            <Text style = {styles.infoLabel}>Following</Text>
                        </TouchableOpacity>
        })
    }
};

let tabNavigatorConfig = {
    tabBarPosition: 'bottom',
    animationEnabled: true,
    swipeEnabled: true
};

const MyGitHubNavigator = TabNavigator(routeConfigs, tabNavigatorConfig);

export default MyGitHubNavigator;
