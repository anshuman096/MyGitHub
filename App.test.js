import React from 'react';
import App from './App';
import MyGitHub from './MyGitHub';
import MyRepos from './Repos';
import RepoList from './RepoList';
import MyFollowers from './Followers';
import {FollowersList} from './FollowersList';
import MyFollowing from './Following';
import {FollowingList} from './FollowingList';

import renderer from 'react-test-renderer';


test('Login renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});



test('MyGitHub renders without crashing', () => {
	const rendered = renderer.create(<MyGitHub />).toJSON();
	expect(rendered).toMatchSnapshot();
});


test('RepoList renders without crashing', () => {
	const rendered = renderer.create(<RepoList />).toJSON();
	expect(rendered).toMatchSnapshot();
});



test('FollowersList renders without crashing', () => {
	const rendered = renderer.create(<FollowersList />).toJSON();
	expect(rendered).toMatchSnapshot();
});


test('FollowingList renders without crashing', () => {
	const rendered = renderer.create(<FollowingList />).toJSON();
	expect(rendered).toMatchSnapshot();
});


