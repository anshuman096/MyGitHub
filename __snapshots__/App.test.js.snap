// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FollowersList renders without crashing 1`] = `null`;

exports[`FollowingList renders without crashing 1`] = `null`;

exports[`Login renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="stretch"
    source={
      Object {
        "testUri": "../../../images/background.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "padding": 20,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontFamily": "Avenir",
              "fontSize": 27,
              "fontWeight": "bold",
              "paddingBottom": 25,
            }
          }
        >
           Git My Hub 
        </Text>
        <Image
          source={
            Object {
              "testUri": "../../../images/github-512.png",
            }
          }
          style={
            Object {
              "height": 200,
              "paddingBottom": 50,
              "width": 200,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          blurOnSubmit={true}
          onChangeText={[Function]}
          placeholder="Username"
          selectTextOnFocus={true}
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "black",
              "borderBottomWidth": 1,
              "borderRadius": 8,
              "fontSize": 18,
              "paddingTop": 20,
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          onEnter={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          selectTextOnFocus={true}
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "black",
              "borderBottomWidth": 1,
              "borderRadius": 8,
              "fontSize": 18,
              "paddingTop": 20,
            }
          }
        />
        <View
          style={
            Object {
              "margin": 7,
            }
          }
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#f5f5f5",
              "borderRadius": 4,
              "opacity": 1,
              "padding": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            Login
          </Text>
        </View>
        <View
          style={
            Object {
              "margin": 7,
            }
          }
        />
        <ActivityIndicator
          animating={false}
          color="#0000ff"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`MyGitHub renders without crashing 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    automaticallyAdjustContentInsets={false}
    bounces={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    overScrollMode="never"
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
        testID={undefined}
      >
        <View
          collapsable={false}
          removeClippedSubviews={false}
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <ActivityIndicator
              animating={true}
              color="#0000ff"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
        </View>
      </View>
      <View
        style={null}
        testID={undefined}
      />
      <View
        style={null}
        testID={undefined}
      />
      <View
        style={null}
        testID={undefined}
      />
    </View>
  </RCTScrollView>
  <View
    collapsable={undefined}
    style={undefined}
  >
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#F7F7F7",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#a9a9a9",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              Profile
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#a9a9a9",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              Repos
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#a9a9a9",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              Followers
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flexGrow": 1,
              }
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            />
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#a9a9a9",
                  "fontFamily": "Avenir",
                  "fontSize": 18,
                  "paddingBottom": 10,
                  "paddingTop": 10,
                }
              }
            >
              Following
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`RepoList renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
